### PR TITLE
fix redirect after login

### PR DESCRIPTION
**Pull request by anbraten**
fix a comment https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1027716962

### CODE REVIEW

**Comment by 6543** [comment]
@anbraten does it fix https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1027716962 ?

File: web/src/views/repo/RepoWrapper.vue [hunk 1]
```diff
@@ -83,6 +85,12 @@ export default defineComponent({
     provide('builds', builds);
 
     async function loadRepo() {
+      // TODO: remove after we fixed #485 as we should then be able to have a look on public repos as well
+      if (!isAuthenticated) {
+        await router.replace({ name: 'login', query: { url: route.fullPath } });
+        return;
+      }
```
Comment (anbraten) [inline web/src/views/repo/RepoWrapper.vue:92]:
```suggestion
      // TODO: remove after we fixed #485 as we should then be able to have a look on public repos as well
      /* if (!isAuthenticated) {
        await router.replace({ name: 'login', query: { url: route.fullPath } });
        return;
      } */
```

**Comment by 6543** [comment]
oauth broke :/

**Comment by 6543** [comment]
oauth broken :/
Screenshot at 2022-03-24 02-58-17

**Comment by anbraten** [comment]
@6543 Are you sure your error is related to this PR? I can't reproduce it.

**Comment by 6543** [comment]
yes same test instance, if i use master branch it works if i jump to this branch it fails.

I'll try to collect more info for  you since you can not reproduce it

**Comment by anbraten** [comment]
@6543 I tested again using my local gitea setup and it works just perfectly. I don't see how the changes could even lead to your error as they do not really touch the oauth setup at all 🤔 

https://user-images.githubusercontent.com/6918444/166634649-edfdc14a-eb64-46fe-b57a-47167ea00852.mp4

**Comment by anbraten** [comment]
> @anbraten https://cloud.obermui.de/s/T3jEYpPeo9b9yc6

Url looks fine. The error you have in your video at the beginning was expected. Should we remove it? Apart from that could you do the login and are you send to the page initially requested or are you still getting this  `OAuth Error` you posted before?

**Comment by 6543** [comment]
@anbraten it s a public repo so i expected no error

**Comment by anbraten** [comment]
@6543 Do you have public in the project settings of the woodpecker project? For me this works fine.

**Comment by 6543** [comment]
is_private=false

**Comment by anbraten** [comment]
> is_private=false

Which prop do you refer to? Repos in WP have `repo_visibility` which handles the visibility for WP (needs  to be `public` to see repo project logs / data without login) and we have  `repo_private` (`IsSCMPrivate`) which refers to the visibility of the repo in your forge which isn't used for the WP authentication only for clone at least if I am not mixing sth up here.

**Comment by 6543** [comment]
```
MariaDB [oo]> select repo_private,repo_visibility from repos where repo_full_name like 'hahaO/testCIservices';
+--------------+-----------------+
| repo_private | repo_visibility |
+--------------+-----------------+
|            0 |                 |
+--------------+-----------------+
1 row in set (0.000 sec)
```

**Comment by 6543** [comment]
I'll try to investigate on my own ...

**Review by 6543 — APPROVED** [review]

**Pull request merged**